module/Finna/src/Finna/Controller/MyResearchController.php

```diff
@@ -1347,6 +1348,115 @@ public function importAction()
+        $functionConfig = $catalog->checkFunction(
+            'getMyTransactionHistory',
+            $patron
+        );
+        if (false === $functionConfig) {
+            $this->flashMessenger()->addErrorMessage('ils_action_unavailable');
+            return $this->createViewModel();
+        }
+
+        $recordLoader = $this->serviceLocator->get(\VuFind\Record\Loader::class);
+        $page = 1;
+        $history = [];
+        do {
+            // Try to use large page size, but take ILS limits into account
+            $pageOptions = $this->getPaginationHelper()
+                ->getOptions($page, null, 1000, $functionConfig);
+            $result = $catalog
+                ->getMyTransactionHistory($patron, $pageOptions['ilsParams']);
+
+            if (isset($result['success']) && !$result['success']) {
+                $this->flashMessenger()->addErrorMessage($result['status']);
+                return $this->redirect()->toRoute('checkouts-history');
+            }
+
+            $ids = [];
+            foreach ($result['transactions'] as $current) {
+                $id = $current['id'] ?? '';
+                $source = $current['source'] ?? DEFAULT_SEARCH_BACKEND;
+                $ids[] = compact('id', 'source');
+            }
+            $records = $recordLoader->loadBatch($ids, true);
+            foreach ($result['transactions'] as $i => $current) {
+                $driver = $records[$i];
+                $format = $driver->getFormats();
+                $format = end($format);
+                $author = $driver->tryMethod('getNonPresenterAuthors');
+
+                $loan = [];
+                $loan[] = $current['title'] ? '"' . $current['title'] . '"' : '';
+                $loan[] = $this->translate($format);
+                $loan[] = $author[0]['name'] ?? '';
+                $loan[] = $current['publication_year'] ?? '';
+                $loan[] = empty($current['borrowingLocation'])
+                    ? ''
+                    : $this->translateWithPrefix('location_', $current['borrowingLocation']);
+                $loan[] = $current['checkoutDate'] ?? '';
+                $loan[] = $current['returnDate'] ?? '';
+                $loan[] = $current['dueDate'] ?? '';
+
+                $history[] = $loan;
+            }
+            $pageEnd = $pageOptions['ilsPaging']
+                ? ceil($result['count'] / $pageOptions['limit'])
+                : 1;
+            $page++;
+        } while ($page <= $pageEnd);
+        $response = $this->getResponse();
+        $response->getHeaders()
+            ->addHeaderLine('Content-Type', 'text/csv')
+            ->addHeaderLine(
+                'Content-Disposition',
+                'attachment; filename="finna-loan-history.csv'
+            );
+
+        try {
+            $outputPath = tempnam('/tmp', 'csv');
+            $handle = fopen($outputPath, 'w');
+            // UTF-8 BOM
+            fwrite($handle, chr(0xEF) . chr(0xBB) . chr(0xBF));
```

Comment:
CSV:ssä ei pitäisi olla BOM:ia. Jos se on sitä varten, että Excel tajuaa muodon, niin parempi olisi tehdä suoraan Excel-muotoon.

Reply:
Jeps, Excelin takia laitoin kun ääkköset ei näkyny oikein. Otin nyt pois, onko siis tarvetta tehä jotain muuta muutoksia tähän?

Reply:
Joo, Excel on hölmö, mutta BOM ei silti kuulu UTF-8:iin. Ehdottaisin, että tehdään tämä kerralla kunnolla esim. niin, että käyttäjä voi valita muodon käyttötarkoituksen mukaan. [PhpSpreadsheet](https://phpspreadsheet.readthedocs.io/en/latest/):llä saisi saman tien kattavan joukon vientimuotoja, ja esim. päivämäärät saisi laitettua [kunnolla](https://phpspreadsheet.readthedocs.io/en/latest/topics/The%20Dating%20Game/).